service/src/main/kotlin/app/cash/backfila/service/runner/BackfillRunner.kt

```diff
@@ -130,17 +133,26 @@ class BackfillRunner private constructor(
         val dbRunning = factory.transacter.transaction { session ->
```

Comment:
We will miss a bit of progress between the delay and we exit here.

Reply:
maybe we delay at the start of the loop instead of the end. I think that fixes it.

Reply:
let me think about this

Reply:
yeah there is a potential loss. why don't we just save on exit?

Reply:
You can but be very careful, you might have lost the lease.

Reply:
You'd basically just be duplicating the logic in that loop. I think it actually makes sense to move the wait to the top of the loop. When you enter the loop the lease is rather recent anyway.

Reply:
yeah that makes sense